Sources/PackageLoading/PackageBuilder.swift

```diff
@@ -835,20 +835,36 @@ public final class PackageBuilder {
         return conditions
     }
 
+    // When building tests on macOS, we are matching the minimum deployment target of the XCTest framework.
+    func xcTestMinimumDeploymentTarget(for platform: PackageModel.Platform) -> PlatformVersion {
+        if let minDeploymentTarget = Self._xcTestMinimumDeploymentTarget[platform] {
+            return minDeploymentTarget
+        }
+        Self._xcTestMinimumDeploymentTarget[platform] = computeXCTestMinimumDeploymentTarget(for: platform)
```

Comment:
this should be input to PackageBuilder as it's not supposed to have any side effects.

Reply:
Ah yah, I missed this before 😢